app/src/main/java/com/github/se/wanderpals/service/MapManager.kt

```diff
@@ -0,0 +1,186 @@
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.AutocompletePrediction
+import com.google.android.libraries.places.api.model.AutocompleteSessionToken
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.model.RectangularBounds
+import com.google.android.libraries.places.api.net.FetchPlaceRequest
+import com.google.android.libraries.places.api.net.FetchPlaceResponse
+import com.google.android.libraries.places.api.net.FindAutocompletePredictionsRequest
+import com.google.android.libraries.places.api.net.FindAutocompletePredictionsResponse
+import com.google.android.libraries.places.api.net.PlacesClient
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Class to manage the map and location services.
+ *
+ * @param context The context of the activity.
+ */
+class MapManager(private val context: Context) {
+
+  private val placeFields =
+      listOf(
+          Place.Field.LAT_LNG,
+          Place.Field.NAME,
+          Place.Field.ADDRESS,
+          Place.Field.BUSINESS_STATUS,
+          Place.Field.PHONE_NUMBER,
+          Place.Field.RATING,
+          Place.Field.USER_RATINGS_TOTAL,
+          Place.Field.WEBSITE_URI,
+          Place.Field.CURRENT_OPENING_HOURS,
+          Place.Field.ICON_URL)
+
+  private lateinit var fusedLocationClient: FusedLocationProviderClient
+  private lateinit var placesClient: PlacesClient
+
+  private lateinit var locationPermissionRequest: ActivityResultLauncher<Array<String>>
+
+  private var startingLocation: LatLng = LatLng(46.519653, 6.632273)
+
+  /** Function to initialize the Places API and the FusedLocationProviderClient. */
+  fun initClients() {
+    Places.initialize(context, BuildConfig.MAPS_API_KEY)
+    placesClient = Places.createClient(context)
+    fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)
+  }
+
+  /**
+   * Function to get the starting location of the map.
+   *
+   * @return The starting location of the map.
+   */
+  fun getStartingLocation(): LatLng {
+    return startingLocation
+  }
+
+  /**
+   * Function to change the starting location of the map.
+   *
+   * @param newLocation The new location to set as the starting location.
+   */
+  fun changeStartingLocation(newLocation: LatLng) {
+    startingLocation = newLocation
+  }
+
+  /**
+   * Function to set the permission request for location.
+   *
+   * @param request The permission request for location.
+   */
+  fun setPermissionRequest(request: ActivityResultLauncher<Array<String>>) {
+    locationPermissionRequest = request
+  }
+
+  /** Function to ask for location permission. */
+  fun askLocationPermission() {
+    if (::locationPermissionRequest.isInitialized) {
+      locationPermissionRequest.launch(
+          arrayOf(
+              Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION))
```

Comment:
Great that the app gives the user the choice, of what they wish to share :+1: 
Very security focused!